### PR TITLE
BinaryFormatter Migration Guide: Formatting, exception type, API links

### PR DESCRIPTION
**Pull request by jeffhandley**
For the [BinaryFormatter Migration Guide](https://aka.ms/binaryformatter-migration-guide):

1. Specify the exception that is thrown (`PlatformNotSupportedException`)
2. Change WinForms/WPF "applications" to "apps"
3. Add spacing between bulleted lines on the MessagePack page (matching the System.Text.Json page's formatting to give space between each statement in the list)
4. Add API docs links, especially to the `NrbfDecoder` APIs
5. Fix broken links to the BinaryFormatter security guide

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/standard/serialization/binaryformatter-migration-guide/choose-a-serializer.md](https://github.com/dotnet/docs/blob/42575322ae51d06453978da71e694e40866898d1/docs/standard/serialization/binaryformatter-migration-guide/choose-a-serializer.md) | [docs/standard/serialization/binaryformatter-migration-guide/choose-a-serializer](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/binaryformatter-migration-guide/choose-a-serializer?branch=pr-en-us-42088) |
| [docs/standard/serialization/binaryformatter-migration-guide/functionality-reference.md](https://github.com/dotnet/docs/blob/42575322ae51d06453978da71e694e40866898d1/docs/standard/serialization/binaryformatter-migration-guide/functionality-reference.md) | [docs/standard/serialization/binaryformatter-migration-guide/functionality-reference](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/binaryformatter-migration-guide/functionality-reference?branch=pr-en-us-42088) |
| [docs/standard/serialization/binaryformatter-migration-guide/index.md](https://github.com/dotnet/docs/blob/42575322ae51d06453978da71e694e40866898d1/docs/standard/serialization/binaryformatter-migration-guide/index.md) | [docs/standard/serialization/binaryformatter-migration-guide/index](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/binaryformatter-migration-guide/index?branch=pr-en-us-42088) |
| [docs/standard/serialization/binaryformatter-migration-guide/migrate-to-messagepack.md](https://github.com/dotnet/docs/blob/42575322ae51d06453978da71e694e40866898d1/docs/standard/serialization/binaryformatter-migration-guide/migrate-to-messagepack.md) | [docs/standard/serialization/binaryformatter-migration-guide/migrate-to-messagepack](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/binaryformatter-migration-guide/migrate-to-messagepack?branch=pr-en-us-42088) |
| [docs/standard/serialization/binaryformatter-migration-guide/read-nrbf-payloads.md](https://github.com/dotnet/docs/blob/42575322ae51d06453978da71e694e40866898d1/docs/standard/serialization/binaryformatter-migration-guide/read-nrbf-payloads.md) | [docs/standard/serialization/binaryformatter-migration-guide/read-nrbf-payloads](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/binaryformatter-migration-guide/read-nrbf-payloads?branch=pr-en-us-42088) |
| [docs/standard/serialization/binaryformatter-migration-guide/winforms-applications.md](https://github.com/dotnet/docs/blob/42575322ae51d06453978da71e694e40866898d1/docs/standard/serialization/binaryformatter-migration-guide/winforms-applications.md) | [docs/standard/serialization/binaryformatter-migration-guide/winforms-applications](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/binaryformatter-migration-guide/winforms-applications?branch=pr-en-us-42088) |
| [docs/standard/serialization/binaryformatter-migration-guide/winforms-wpf-ole-guidance.md](https://github.com/dotnet/docs/blob/42575322ae51d06453978da71e694e40866898d1/docs/standard/serialization/binaryformatter-migration-guide/winforms-wpf-ole-guidance.md) | [docs/standard/serialization/binaryformatter-migration-guide/winforms-wpf-ole-guidance](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/binaryformatter-migration-guide/winforms-wpf-ole-guidance?branch=pr-en-us-42088) |
| [docs/standard/serialization/binaryformatter-migration-guide/wpf-applications.md](https://github.com/dotnet/docs/blob/42575322ae51d06453978da71e694e40866898d1/docs/standard/serialization/binaryformatter-migration-guide/wpf-applications.md) | [docs/standard/serialization/binaryformatter-migration-guide/wpf-applications](https://review.learn.microsoft.com/en-us/dotnet/standard/serialization/binaryformatter-migration-guide/wpf-applications?branch=pr-en-us-42088) |

### CODE REVIEW

**Review by terrajobst — APPROVED** [review]
Thanks for fixing those, looks great!

**Comment by jeffhandley** [comment]
OK -- the links to the binaryformatter-security-guide are finally correct. It required:

1. Absolute link starting with `/dotnet/standard/...`
2. Removing the `.md` file extension

File: docs/standard/serialization/binaryformatter-migration-guide/read-nrbf-payloads.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ helpviewer_keywords:
...
 
-As part of .NET 9, a new `NrbfDecoder` class was introduced to decode NRBF payloads without performing _deserialization_ of the payload. This API can safely be used to decode trusted or untrusted payloads without any of the risks that BinaryFormatter deserialization carries. However, `NrbfDecoder` merely decodes the data into structures an application can further process. Care must be taken when using `NrbfDecoder` to safely load the data into the appropriate instances.
+As part of .NET 9, a new [`NrbfDecoder`](/dotnet/api/system.formats.nrbf.nrbfdecoder) class was introduced to decode NRBF payloads without performing _deserialization_ of the payload. This API can safely be used to decode trusted or untrusted payloads without any of the risks that BinaryFormatter deserialization carries. However, `NrbfDecoder` merely decodes the data into structures an application can further process. Care must be taken when using `NrbfDecoder` to safely load the data into the appropriate instances.
```
Comment (BillWagner) [inline docs/standard/serialization/binaryformatter-migration-guide/read-nrbf-payloads.md:19]: Should this be an xref instead?

File: docs/standard/serialization/binaryformatter-migration-guide/read-nrbf-payloads.md [hunk 2]
```diff
@@ -42,10 +42,10 @@ When using `NrbfDecoder`, it is important not to reintroduce those capabilities
 
 ### Identify NRBF payloads
 
-`NrbfDecoder` provides two `StartsWithPayloadHeader` methods that let you check whether a given stream or buffer starts with the NRBF header. It's recommended to use these methods when you're migrating payloads persisted with `BinaryFormatter` to a [different serializer](./choose-a-serializer.md):
+`NrbfDecoder` provides two [`StartsWithPayloadHeader`](/dotnet/api/system.formats.nrbf.nrbfdecoder.startswithpayloadheader) methods that let you check whether a given stream or buffer starts with the NRBF header. It's recommended to use these methods when you're migrating payloads persisted with `BinaryFormatter` to a [different serializer](./choose-a-serializer.md):
```
Comment (BillWagner) [inline docs/standard/serialization/binaryformatter-migration-guide/read-nrbf-payloads.md:45]: Also should be an xref

File: docs/standard/serialization/binaryformatter-migration-guide/read-nrbf-payloads.md [hunk 3]
```diff
@@ -42,10 +42,10 @@ When using `NrbfDecoder`, it is important not to reintroduce those capabilities
...
 
-- Check if the payload read from storage is an [NRBF](/openspecs/windows_protocols/ms-nrbf/) payload.
-- If so, read it with `NrbfDecoder`, serialize it back with a new serializer, and overwrite the data in the storage.
+- Check if the payload read from storage is an [NRBF](/openspecs/windows_protocols/ms-nrbf/) payload with [`NrbfDecoder.StartsWithPayloadHeader`](/dotnet/api/system.formats.nrbf.nrbfdecoder.startswithpayloadheader).
+- If so, read it with [`NrbfDecoder.Decode`](/dotnet/api/system.formats.nrbf.nrbfdecoder.decode), serialize it back with a new serializer, and overwrite the data in the storage.
```
Comment (BillWagner) [inline docs/standard/serialization/binaryformatter-migration-guide/read-nrbf-payloads.md:48]: same.

**Review by BillWagner — COMMENTED** [review]
This overall looks good.

I think it would be better to use the xref style links rather than the page links for API reference.

**Comment by terrajobst** [comment]
@BillWagner I've replaced all use of links to API with `xref`.

**Pull request merged**